examples/pytorch/graphsage/train_full.py

```diff
@@ -1,171 +1,93 @@
-        loss = F.cross_entropy(logits[train_nid], labels[train_nid])
-
+        loss = loss_fcn(logits[train_mask], labels[train_mask])
         optimizer.zero_grad()
         loss.backward()
         optimizer.step()
-
-        if epoch >= 3:
-            dur.append(time.time() - t0)
-
-        acc = evaluate(model, g, features, labels, val_nid)
-        print("Epoch {:05d} | Time(s) {:.4f} | Loss {:.4f} | Accuracy {:.4f} | "
-              "ETputs(KTEPS) {:.2f}".format(epoch, np.mean(dur), loss.item(),
-                                            acc, n_edges / np.mean(dur) / 1000))
-
-    print()
-    acc = evaluate(model, g, features, labels, test_nid)
-    print("Test Accuracy {:.4f}".format(acc))
-
+        acc = evaluate(g, features, labels, val_mask, model)
+        print("Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} "
+              . format(epoch, loss.item(), acc))
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='GraphSAGE')
-    register_data_args(parser)
-    parser.add_argument("--dropout", type=float, default=0.5,
-                        help="dropout probability")
-    parser.add_argument("--gpu", type=int, default=-1,
-                        help="gpu")
-    parser.add_argument("--lr", type=float, default=1e-2,
-                        help="learning rate")
-    parser.add_argument("--n-epochs", type=int, default=200,
-                        help="number of training epochs")
-    parser.add_argument("--n-hidden", type=int, default=16,
-                        help="number of hidden gcn units")
-    parser.add_argument("--n-layers", type=int, default=1,
-                        help="number of hidden gcn layers")
-    parser.add_argument("--weight-decay", type=float, default=5e-4,
-                        help="Weight for L2 loss")
-    parser.add_argument("--aggregator-type", type=str, default="gcn",
-                        help="Aggregator type: mean/gcn/pool/lstm")
+    parser.add_argument("--dataset", type=str, default="cora",
+                        help="Dataset name ('cora', 'citeseer', 'pubmed').")
     args = parser.parse_args()
-    print(args)
-
-    main(args)
+    print(f'Training with DGL built-in GraphSage module.')
+    
+    # load and preprocess dataset
+    transform = AddSelfLoop()  # by default, it will first remove self-loops to prevent duplication
+    if args.dataset == 'cora':
+        data = CoraGraphDataset(transform=transform)
+    elif args.dataset == 'citeseer':
+        data = CiteseerGraphDataset(transform=transform)
+    elif args.dataset == 'pubmed':
+        data = PubmedGraphDataset(transform=transform)
+    else:
+        raise ValueError('Unknown dataset: {}'.format(args.dataset))
+    g = data[0]
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    g = g.int().to(device)
+    features = g.ndata['feat']
+    labels = g.ndata['label']
+    masks = g.ndata['train_mask'], g.ndata['val_mask'], g.ndata['test_mask']
```

Comment:
Maybe just do `masks = g.ndata['train_mask'], g.ndata['val_mask']` and pass `g.ndata['test_mask']` directly at L92.